spec/unit/attributes/update_spec.rb

```diff
@@ -3,119 +3,201 @@
+          end
+          it { expect(subject['emb_one']).to eq [{ 'em_foo' => 'Em-Foo' }, {}] }
         end
-        it { expect(subject['emb_one']).to eq [{}, { 'em_foo' => 'Em-Foo-new', 'em_bar' => 'Em-Bar-new' }] }
-      end
 
-      context 'when new value soft-deleted' do
-        before(:each) { ModelOne.track_history on: :emb_one }
-        let(:changes) do
-          { 'emb_one' => [{ 'em_foo' => 'Em-Foo' }, { 'em_foo' => 'Em-Foo-new', 'deleted_at' => Time.now }] }
+        context 'when not tracked' do
+          before(:each) do
+            ModelOne.track_history on: :fields
+            allow(ModelOne).to receive(:dynamic_enabled?) { false }
+          end
+          it { expect(subject['emb_one']).to be_nil }
+        end
+
+        after(:all) do
+          Object.send(:remove_const, :ModelOne)
+          Object.send(:remove_const, :EmbOne)
         end
-        it { expect(subject['emb_one']).to eq [{ 'em_foo' => 'Em-Foo' }, {}] }
       end
 
-      context 'when not tracked' do
+      context 'Case: relation with alias' do
+        before(:all) do
+          class ModelOne
+            include Mongoid::Document
+            include Mongoid::History::Trackable
+            store_in collection: :model_ones
+            embeds_one :emb_one, store_as: :eon
+            track_history on: :fields
+          end
+
+          class EmbOne
+            include Mongoid::Document
+            field :em_foo
+            field :em_bar
+            embedded_in :model_one
+          end
+        end
+
         before(:each) do
-          ModelOne.track_history on: :fields
-          allow(ModelOne).to receive(:dynamic_enabled?) { false }
+          ModelOne.clear_trackable_memoization
+          ModelOne.track_history on: :emb_one
+          allow(base).to receive(:changes) { changes }
         end
-        it { expect(subject['emb_one']).to be_nil }
-      end
 
-      after(:all) do
-        Object.send(:remove_const, :ModelOne)
-        Object.send(:remove_const, :EmbOne)
+        let(:obj_one) { ModelOne.new }
+        let(:base) { described_class.new(obj_one) }
+        let(:changes) do
+          { 'emb_one' => [{ 'em_foo' => 'Em-Foo' }, { 'em_foo' => 'Em-Foo-new', 'em_bar' => 'Em-Bar-new' }] }
+        end
+        subject { base.attributes }
+        it { expect(subject['eon']).to eq [{ 'em_foo' => 'Em-Foo' }, { 'em_foo' => 'Em-Foo-new', 'em_bar' => 'Em-Bar-new' }] }
+
+        after(:all) do
+          Object.send(:remove_const, :ModelOne)
+          Object.send(:remove_const, :EmbOne)
+        end
       end
     end
 
     describe '#insert_embeds_many_changes' do
-      before(:all) do
-        class ModelOne
-          include Mongoid::Document
-          include Mongoid::History::Trackable
-          store_in collection: :model_ones
-          embeds_many :emb_ones
-          track_history on: :fields
+      context 'Case: relation without alias' do
+        before(:all) do
+          class ModelOne
```

Comment:
Ah yeah, you have to `const_set` the name. I suppose if you generate a random one for the class name that works, but does seem like a lot of work.